Run provisioning in Travis CI

sauloperez opened the pull request:
The cost-effective solution we came with to have continuous integration for our provisioning scripts works as follows:

* Executes Ansible using a local connection, modifying the same CI container
it's running on.
* Because of that, it removes any installed PostgreSQL versions. This
way, we ensure our own database role works.

Also, we had to manually remove PostgreSQL cluster directories. The `--purge` apt's argument doesn't seem to do it. I tried on a LXC and running `sudo apt remove --purge postgresql-9.4` opens a ncurses-like UI to confirm the actual purge of cluster directories. I guess this is what we can't do when executing it in CI.

Finally, we add an argument to create pg_hba.conf file if it doesn't exist. It was necessary to make the job succeed.

sauloperez committed:
Run provisioning in Travis CI

The cost-effective solution we came with to have continuous integration
for our provisioning scripts works as follows:

* Executes Ansible using a local connection, modifying the same CI container
it's running on.
* Because of that, it removes any installed PostgreSQL versions. This
way, we ensure our own database role works.

Also, we had to manually remove PostgreSQL cluster directories.

The `--purge` apt's argument doesn't seem to do it. I tried on a LXC and
running `sudo apt remove --purge postgresql-9.4` opens a ncurses-like UI
to confirm the actual purge of cluster directories. I guess this is what
we can't do when executing it in CI.

Finally, we add an argument to create pg_hba.conf file if it doesn't
exist. It was necessary to make the job succeed.

raneq commented:
Regarding apt purge:
> Also, we had to manually remove PostgreSQL cluster directories. The --purge apt's argument doesn't seem to do it. I tried on a LXC and running sudo apt remove --purge postgresql-9.4 opens a ncurses-like UI to confirm the actual purge of cluster directories. I guess this is what we can't do when executing it in CI.

You can try with `apt purge --yes`. The relevant part of manpages:


`man apt-get`
>        -y, --yes, --assume-yes
>            Automatic yes to prompts; assume "yes" as answer to all prompts and run non-interactively. If an undesirable situation, such as changing a held
>            package, trying to install an unauthenticated package or removing an essential package occurs then apt-get will abort. Configuration Item:
>            APT::Get::Assume-Yes


>     purge
>           purge is identical to remove except that packages are removed and purged (any configuration files are deleted too).


>        --purge
>           Use purge instead of remove for anything that would be removed. An asterisk ("*") will be displayed next to packages which are scheduled to be
>           purged.  remove --purge is equivalent to the purge command. Configuration Item: APT::Get::Purge.

sauloperez commented:
That's exactly what I tried (appending a -y) as well as using `purge` without any luck. Now that we got it working we could try it again in a separate PR.

enricostano reviewed (approved):
I still don't grasp the difference between this approach and using molecule, a part the fact that with molecule you don't need to care about existing state of the host you provision.

In any case I would go with this and we'll try to have a better understanding in the future.

:clap: :muscle:

raneq commented:
By the way, this tests will be great to include in other roles, thanks!

sauloperez commented:
I just know molecule's name, sorry. But obviously this is just a quick solution. Next step could be starting from a known good state, either a container or a cheap VPS.

sauloperez commented:
@raneq I created https://github.com/coopdevs/timeoverflow-provisioning/issues/95 to keep track of your suggestion.

sauloperez committed:
Do not create the pg_hba.conf if it doesn't exist

This is no longer needed. Because we install PostgreSQL one step above
the file it's always going to be present.
